File: Tema 3 -Sentencia condicional (if-switch)/ejercicios/ejercicio17  primera cifra de un numero entero/archives/17.py
import sys
import math

# Obtener los argumentos de la línea de comandos
b = int(sys.argv[1])
a = abs(b)
if 0 <= a <= 99999:
    if 0 <= a <= 9:
        mensaje = f"El primer digito de {b} <br>es: {a}"
    elif 10 <= a <= 99:
        mensaje = f"El primer digito de {b} <br>es: {int(a/10)}"
    elif 100 <= a <= 999:
        mensaje = f"El primer digito de {b} <br>es: {int(a/100)}"
    elif 1000 <= a <= 9999:
        mensaje = f"El primer digito de {b} <br>es: {int(a/1000)}"
    elif 10000 <= a <= 99999:
        mensaje = f"El primer digito de {b} <br>es: {int(a/10000)}"
else:
    mensaje = "Ingrese un maximo de 5 digitos"
print(mensaje)

File: Tema 3 -Sentencia condicional (if-switch)/ejercicios/ejercicio17  primera cifra de un numero entero/archives/17.php
<?php
$num1 = ($_POST["num1"]);
if (
	isset($num1) &&
	trim($num1) !== "" &&
	is_numeric($num1) &&
	filter_var($num1, FILTER_VALIDATE_INT) !== false
) {
	/* SECTION inicio programa */
	$a = abs($num1);
	if ($a >= 0 && $a <= 99999) {
		switch (true) {
			case ($a <= 9):
				$mensaje = "El primer digito de $num1 <br>es: $a";
				break;
			case ($a <= 99):
				$mensaje = "El primer digito de $num1 <br>es: " . (int)($a / 10);
				break;
			case ($a <= 999):
				$mensaje = "El primer digito de $num1 <br>es: " . (int)($a / 100);
				break;
			case ($a <= 9999):
				$mensaje = "El primer digito de $num1 <br>es: " . (int)($a / 1000);
				break;
			case ($a <= 99999):
				$mensaje = "El primer digito de $num1 <br>es: " . (int)($a / 10000);
				break;
		}
	} else {
		$mensaje = "Solo se aceptan numeros de hasta 5 digitos ";
	}
	/* !SECTION fin programa */
} else {
	$mensaje = "Ingrese un número Entero.";
}
echo $mensaje;
